test.sh

```diff
@@ -5,16 +5,19 @@ set -e
 function finally() {
 	# clean up aka terminate test server
 	if [ $DRY_RUN = false ]; then
-		echo "Terminating server..."
-		aws ec2 terminate-instances --region $region --instance-ids $instanceid > /dev/null
-		delete_test_volumes
-		delete_profile_for_volume_attachment
-		if [ -n "$mint_bucket" ]; then
-			taupageyamlfile="$(pwd)/test-userdata.yaml"
-			echo "change mint-bucket to a example again"
-			sed -i "1,$ s/mint_bucket.*$/mint_bucket:\ S3-MINT-BUCKET/" $taupageyamlfile
+		if [ "$?" -gt 0 ] || [ ${1:-false} = true ]; then
```

Comment:
@dehora well we can leave out it. But i noticed that the behaviour might be different depending on the instance type. Therefore i added the possibility to check different instance types